homeassistant/components/binary_sensor/isy994.py

```diff
@@ -49,22 +70,183 @@ def setup_platform(hass, config: ConfigType,
+        """Add a child node to this binary sensor device.
+
+        The child node can either be a node that receives the 'off' events, or
+        a heartbeat node for reporting that this device is still alive
+        """
+        subnode_id = int(child.nid[-1])
+        if subnode_id == 2:
+            # "Negative" node that can be used to represent a negative state
+            # when it reports "On"
+            self._negative_node = child
+        elif subnode_id == 4:
+            # Heartbeat node that just reports "On" every 24 hours
+            self._heartbeat_timestamp = STATE_UNKNOWN
+            self._heartbeat_node = child
+
+    def _negative_node_control_handler(self, event: object) -> None:
+        """Handle an "On" control event from the "negative" node."""
+        if event == 'DON':
+            _LOGGER.debug("Sensor %s turning Off via the Negative node "
+                          "sending a DON command", self.name)
+            self._computed_state = False
+            self.schedule_update_ha_state()
+
+    def _positive_node_control_handler(self, event: object) -> None:
+        """Handle On and Off control event coming from the primary node.
+
+        Depending on device configuration, sometimes only On events
+        will come to this node, with the negative node representing Off
+        events
+        """
+        if event == 'DON':
+            _LOGGER.debug("Sensor %s turning On via the Primary node "
+                          "sending a DON command", self.name)
+            self._computed_state = True
+            self.schedule_update_ha_state()
+        if event == 'DOF':
+            _LOGGER.debug("Sensor %s turning Off via the Primary node "
+                          "sending a DOF command", self.name)
+            self._computed_state = False
+            self.schedule_update_ha_state()
+
+    def _heartbeat_node_control_handler(self, event: object) -> None:
+        """Update the heartbeat timestamp when an On event is sent."""
+        if event == 'DON':
+            self._heartbeat_timestamp = datetime.now().isoformat()
+            self.schedule_update_ha_state()
+
+    # pylint: disable=unused-argument
+    def on_update(self, event: object) -> None:
+        """Ignore primary node status updates.
+
+        We listen directly to the Control events on all nodes for this
+        device.
+        """
+        pass
+
+    @property
+    def value(self) -> object:
+        """Get the current value of the device.
+
+        Insteon leak sensors set their primary node to On when the state is
+        DRY, not WET, so we invert the binary state if the user indicates
+        that it is a moisture sensor.
+        """
+        if self._computed_state is None:
+            # Do this first so we don't invert None on moisture sensors
+            return None
+
+        try:
+            if self.device_class == 'moisture':
+                return not self._computed_state
+        except AttributeError:
```

Comment:
What object attribute access could lead to attribute error?

Reply:
Fixed - artifact of a refactor